### DISCUSSION TITLE
Elster Merlin 868

### DISCUSSION THREAD

**Discussion by DaveOz1** [wmbusmeters/wmbusmeters#554] [General]
Good Day,

I'm hoping to be able to decode data received from Merlin 868 LoRa units.

We receive the data from an MQTT server and I can look at the payload in NodeRed. I had hoped that I could send the received payload to wmbusmeters for decoding and then process the results further. Unfortunately I have had no success doing this from the command line. 

I'm a bit lost as to where to start with this. I would be grateful for any pointers and, perhaps even some sample data that will work with the command line, so that I can get an idea of what I'm looking for.

Here's an example payload as received from the MQTT server by NodeRed:
2ac1000000000000430000000000000000000000000000000000000000000000000000000000000000ffff000009000a55138b

I presume that it's encrypted and I do have the keys.

Any guidance to the next step would be gratefully received.

Thanks

**Comment by weetmuts** [comment]
The payload does not look encrypted, way too many zeroes in there. Unfortunately it does not look like (w)mbus either.
Can you explain a bit more how you received the lora telegram? Through a gateway operator that relays telegrams to your own server?

I have started to implement direct lora support, the first lora dongle is partially implemented, iu880b. This is still work in progress though. But it might be interesting to implement a server port to reveice lorawan telegram as well.....

**Reply by DaveOz1** [reply]
Thanks for getting back to me on this.

After discussions within our organisation and with Honeywell, we have learned that we can be given the data structure, but only under a non-disclosure agreement!

All I did to receive the telegram was to configure our LoRaWan provider's gateway (Kerlink) to negotiate a connection with the meter using the usual set of keys. Once that was achieved, I set the gateway to to an HTTP "Push" to my MQTT broker. From there I used NodeRed to grab the resulting payload.

I suppose that once we get the data structure I should be able to parse it and make use of it, but I won't be able to share it with anyone because of the NDA!

It doesn't much help the debugging effort that these things only report-in every 12 hours :-)

Kind Regards

**Comment by weetmuts** [comment]
Is it this clip-on device? https://www.smartenergy.honeywell.com/product/merlin-868-hb-g3/

**Reply by DaveOz1** [reply]
That's the very chap!

My understanding is that it's made by Birdz, who own the intellectual property.

**Reply by MO13456** [reply]
> Is it this clip-on device? https://www.smartenergy.honeywell.com/product/merlin-868-hb-g3/

Any news or guide how to read merlin 868? Thank you!

**Reply by DaveOz1** [reply]
Sadly not.
I'm told that the protocol is only released to end users under a non-disclosure agreement. So that brought everything to a stop!
